### PR TITLE
add urql and refactor the meta query to be paginated

### PR DESCRIPTION
**Pull request by dagda1**
## Motivation

We need to make our queries lazy loaded for large simulated graphs.

## Approach

Introduce [urql](https://github.com/FormidableLabs/urql) on the client and refactor the `meta`  query to be paginated.

The `all` query can be paginated in a subsequent PR.

### CODE REVIEW

File: cli/relay.config.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default {
```
Comment (taras) [inline cli/relay.config.ts]: Do we still need this file?

**Review by taras — COMMENTED** [review]
One question in the comments. In a follow-up PR, I can help to configure the frontend to generate a TypeScript type for the query.

File: cli/relay.config.ts (hunk 1 above)
Comment (dagda1) [reply cli/relay.config.ts]: good spot.

**Comment by dagda1** [comment]
> One question in the comments. In a follow-up PR, I can help to configure the frontend to generate a TypeScript type for the query.

I've removed the spurious files, and any help would be great.  I'm no relay expert.

**Comment by cowboyd** [comment]
The number of total types should generally be low, do we need to paginate it? Or should we start with the Vertex query which will definitely need to be paginated?

**Comment by dagda1** [comment]
> The number of total types should generally be low, do we need to paginate it? Or should we start with the Vertex query which will definitely need to be paginated?

Probably the right thing to do, I just wanted to get something simple working to see what problems urql created which were none.

I'll un-paginate it.

**Review by taras — APPROVED** [review]

**Pull request merged**